base_tier_validation/models/tier_validation.py

```diff
@@ -246,18 +246,22 @@ def _get_validation_exceptions(self, extra_domain=None, add_base_exceptions=True
         """Return Tier Validation Exception field names that matchs custom domain."""
         exception_fields = (
             self.env["tier.validation.exception"]
+            .sudo()
             .search(
                 [
                     ("model_name", "=", self._name),
                     ("company_id", "in", [False] + self.env.company.ids),
                     *(extra_domain or []),
+                    "|",
+                    ("group_ids", "in", self.env.user.groups_id.ids),
+                    ("group_ids", "=", False),
```

Comment:
issue: for security, always append extra domains at the end:
```suggestion
                    "|",
                    ("group_ids", "in", self.env.user.groups_id.ids),
                    ("group_ids", "=", False),
                    *(extra_domain or []),
```

This avoids the case of dangling operands that alter the meaning of the rest of the domain.